src/datasets/load.py

```diff
@@ -435,11 +439,14 @@ def infer_module_for_data_files_in_archives(
     data_files_list: DataFilesList, use_auth_token: Optional[Union[bool, str]]
 ) -> Optional[str]:
     archived_files = []
-    for filepath in data_files_list:
+    for filepath in data_files_list[: config.DATA_FILES_MAX_NUMBER]:
```

Comment:
Here since each glob is costly, it's good to only have 10 at maximum

Reply:
I have added a new config variable specific to limit the number of globbed files.

Reply:
Awesome thanks !